### DISCUSSION TITLE
Enable JetBrains plugin on all IDEs

### DISCUSSION THREAD

**Discussion by Sciguymjm** [community/community#6626] [Copilot Conversations]
Currently in CLion 2021.2.3 I receive [an error](https://i.imgur.com/uZLr294.png) when adding the plugin repo, presumably because there are only certain builds enabled. It works fine in PyCharm. I would like to experiment with this plugin in other IDEs other than Intellij or PyCharm.

**Comment by superman66** [comment]
I'm using WebStorm with the same issue

**Reply by linktlh** [reply]
Make sure you update webstorm, version 212.5457.55 can install it. [Example](https://i.imgur.com/fky3HKc.png)

**Reply by superman66** [reply]
Thanks, It works when i update the latest version.

**Comment by A-z-z-y** [comment]
Getting the same problem in Rider

**Reply by edivanteixeira** [reply]
Me too, I already update all jetbrains programs.

**Comment by linktlh** [comment]
Make sure whatever IDE you're using you **fully update**. Pretty much only the most recent versions of the Jetbrains IDEs are supported. 

To update your IDE go into Settings [CTRL+ALT+S] -> Appearance & Behavior -> System Settings -> Updates -> Check for Updates

**Reply by A-z-z-y** [reply]
I am on the latest version, and after downloading the EAP version of JetBrains Rider (RD-213.5281.17), the repo is still showing errors ("No plugins found. Check the log file for errors.") and therefor not appearing in the plugin marketplace.

**Reply by linktlh** [reply]
For your particular build [RD-213.5281.17] I do see that it is available [https://plugins.jetbrains.com/plugins/list?channel=super-early-bird&build=ED-213.5281.17](https://plugins.jetbrains.com/plugins/list?channel=super-early-bird&build=ED-213.5281.17)

**Reply by A-z-z-y** [reply]
"Connection Faled"
![image](https://user-images.githubusercontent.com/85721603/138604415-055d5198-f731-4205-9bac-d1670e5e2a7f.png)

**Reply by linktlh** [reply]
The link I shared was basically what occurs when the IDE looks at repos. If you want you can download the zip, and install it manually. Though you'll have to update it manually as well.

**Reply by ecarandang** [reply]
> The link I shared was basically what occurs when the IDE looks at repos. If you want you can download the zip, and install it manually. Though you'll have to update it manually as well.

Did manual install on the latest Rider IDE as well as the EAP version, but have the IDE contains conflicting module error.
![image](https://user-images.githubusercontent.com/7642337/138895821-1637ac19-6dcf-4520-b269-ef1af516ca37.png)

**Comment by A-z-z-y** [comment]
Ahh, alright everyone, I have a solution in the case that you are unable to get the repo to work. Here are the steps:

1. Update your IDE using the instructions above
2. Download the plugin with this link: https://plugins.jetbrains.com/plugin/download?updateId=142849
3. Go to your IDE Settings > Plugins > Settings icon > Install plugin from disk > Select the zip from your downloads
4. Restart your IDE (optional, but it's a good idea)
5. Login with GitHub using the prompt the plugin gives you
6. You're done!

Again, only do this if the repo does not work.

Edit: disclaimer that the repo is better :P

**Reply by linktlh** [reply]
Still recommend installing it via the repo and not via this method. Only do this if you are completely unable to make the repo work, as this will render the jetbrains ide unable to check for updates thus causing you to need to manually download all future updates.

**Reply by A-z-z-y** [reply]
Good point! Just gonna edit the original post to include that...

**Reply by Oyshoboy** [reply]
JetBrains Rider is not supported yet and custom repository URL feature seems to be broken, so that's method is worked for me. But copilot plugins keeps crashing time to time. Temporary moved to VS Code for full experience.

**Reply by A-z-z-y** [reply]
It does occasionally crash, but it works well enough for me, and probably for others. But, the solution I provided above does at least temporarily work, which is good enough :)

**Reply by linktlh** [reply]
Rider is enabled on the plug-in repo, (at least the version mentioned in the last main message in this discussion). Make sure to report any crashing as a bug report with logs.

**Reply by ffMathy** [reply]
The provided link is for Two-Oh, not GitHub CoPilot.

What is the link for the real extension?

**Reply by ffMathy** [reply]
Ah found it. Here it is: https://plugins.jetbrains.com/plugin/download?updateId=142849

**Reply by A-z-z-y** [reply]
oh... odd... the link must have changed? weird... just gonna update the original post :D

**Reply by ffMathy** [reply]
However, the plugin still doesn't work on Rider 😑

**Comment by ffMathy** [comment]
Still no official reply from GitHub? 🤔

**Reply by A-z-z-y** [reply]
Unfortunately no, but I hope there will be soon...

**Comment by FeelsBadMan1** [comment]
> Currently in CLion 2021.2.3 I receive [an error](https://i.imgur.com/uZLr294.png) when adding the plugin repo, presumably because there are only certain builds enabled. It works fine in PyCharm. I would like to experiment with this plugin in other IDEs other than Intellij or PyCharm.

Oh lord why it works for you on pycharm but for me not? it just not shows up and after entering the repolink i only see something around the lines "plugin not found in remp / empty"

**Reply by A-z-z-y** [reply]
Make sure you check for updates, and look again. If you still can't find it, you could try [my solution](https://github.com/github/feedback/discussions/6626#discussioncomment-1528785) to manually install the plugin (you will have to update it by yourself though, it won't auto-update or detect updates this way)

**Comment by shadowmint** [comment]
The version linked in this thread is old; if you're manually installing the plugin you should download the latest version from https://plugins.jetbrains.com/plugin/17718-github-copilot/versions

**Reply by zzxwill** [reply]
By upgrading Goland in the way @A-z-z-y mentioned, I have enabled Copilot plugin to this new release and it worked. Thanks.

**Comment by edcarandang** [comment]
Hi all, not related (or maybe it is) to this topic. For those who were able to make the plugin work with their IDEs. Are you all approved to use Github Copilot? I've applied for the waitlist for some time now, but I'm still not approved. Anything I need to do to get approved?

**Reply by zzxwill** [reply]
It took me two months.

**Reply by ffMathy** [reply]
It's not related in my case. I am already using CoPilot in VS Code.

**Reply by linktlh** [reply]
Yes, you must be approved. The plug-in won't function unless you authorize with GitHub which will check your eligibility.
